astropy/io/fits/column.py

```diff
@@ -764,6 +785,46 @@ def name(col, name):
                 'characters, though it may be fewer if the string '
                 'contains special characters like quotes.')
 
+    @ColumnAttribute('TCTYP')
+    def coord_type(col, coord_type):
+        if coord_type is None:
+            return
+
+        if (not isinstance(coord_type, string_types)
+                or len(coord_type) > 8):
+            raise AssertionError(
+                'Coordinate/axis type must be a string of atmost 8 '
+                'characters.')
+
+    @ColumnAttribute('TCRPX')
+    def coord_ref_point(col, coord_ref_point):
+        if coord_ref_point is None:
+            return
+
+        if not _is_int(coord_ref_point) and not isinstance(coord_ref_point, float):
```

Comment:
For the integer types, this includes `np.integer`, so I think we should do the same for `float`. Indeed, one solution would be to add to `util`:
```
all_real_types = all_integer_types + (float, np.float)
```
But really this just duplicates stuff python has built-in: more sensible is to do
```
# at top
import numbers
# here, just one statement
if a is not None and not isinstance(a, numbers.Real):
   ...
```
This will be true also for integers and for `np.integer` and `np.float` (but not for array scalars)